### ISSUE TITLE
Make ulimit optional or use puppet-systemd for dropin

### ISSUE THREAD

**Issue by laugmanuel** [voxpupuli/puppet-redis#390]
This module currently contains `redis::ulimit` on every run (https://github.com/voxpupuli/puppet-redis/blob/master/manifests/config.pp#L41-L43) which can't be disabled due to the parameter constraints here: https://github.com/voxpupuli/puppet-redis/blob/master/manifests/init.pp#L318

This makes it impossible to add an additional `systemd::dropin` using https://github.com/camptocamp/puppet-systemd as the dropin directory get's enforced by `redis::ulimit` and also `systemd::dropin_file` (= duplicate declaration).

Two possible solutions could be:
- Fix the parameter definition to allow `undef` and therefore skip the containment of `redis::ulimit` -> A user would have to explicitly define the ulimit
- Switch to puppet-systemd for managing the dropin directory and the dropin itself (instead of `file` and `augeas`).

## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: any
- Ruby: any
- Distribution: any
- Module version: any

## How to reproduce (e.g Puppet code you use)
```
class { '::redis':
    bind => ['127.0.0.1', $::facts['default_ipaddress']],
  }

  systemd::dropin_file { 'redis-dropin2.conf':
    unit     => 'redis.service',
    content  => '[Unit]\nDescription=Testdescription',
  }
```
## What are you seeing
```sh
Evaluation Error: Error while evaluating a Resource Statement, Evaluation Error: Error while evaluating a Function Call, Duplicate declaration: File[/etc/systemd/system/redis.service.d] is already declared
```
## What behaviour did you expect instead
Two dropin files managed by puppet

**Comment by ekohl** [comment]
Now that we've dropped sysvinit (42fd85c938bdb2e4b88816a41a4abe93da87e10e) it would be good to rely on the systemd feature for this. There is actually code for it in the service template:

https://github.com/voxpupuli/puppet-redis/blob/551e58b50dd6101b9ad563b81de07aafaa2dfdb1/templates/service_templates/redis.service.erb#L22

However, for the default service we use the system provided one. Using a drop in is good. We should kill `redis::ulimit` and make it part of `redis::instance`.

I would also be fine with making it `Optional[Integer[0]]` so it can be skipped.

**Comment by laugmanuel** [comment]
I'll prepare a PR with fixed parameter constraints and also the use of [puppet-systemd](https://github.com/camptocamp/puppet-systemd) for Dropin management.
Is there any policy for voxpupuli modules regarding external dependencies?

**Comment by ekohl** [comment]
Closing since this was resolved in #396.
